File: C++ progs/Playlist.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    map<int, int> mp;
    int n;
    cin >> n;
    int j = 1, ans = 0;
    for (int i = 1; i <= n; i++)
    {
        int k;
        cin >> k;
        if (mp[k])
        {
            ans = max(ans, i - j);
            j = max(j, mp[k] + 1);
            mp[k] = i;
        }
        else
        {
            ans = max(ans, i - j + 1);
            mp[k] = i;
        }
    }
    ans = max(ans, n - j + 1);
    cout << ans << endl;
    return 0;
}

File: Python progs/Playlist.py
def main():
    # input n
    n = int(input())
    # input the list
    v = list(map(int, input().split()))
    j = 0
    ans = 0
    mp = {}
    for i in range(n):
        if(v[i] in mp):
            ans = max(ans,i - j - 1)
            j = max(j, mp[v[i]])
            mp[v[i]] = i
        else:
            ans = max(ans, i - j)
            mp[v[i]] = i
    ans = max(ans, n - j - 1)
    print(ans)



if __name__ == "__main__":
    main()
